src/components/Notification/Notification.tsx

```diff
@@ -0,0 +1,175 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React, { useEffect, useRef } from "react";
+import type { HTMLProps, ReactNode } from "react";
+
+import Button, { ButtonAppearance } from "../Button";
+import type { Props as ButtonProps } from "../Button";
+
+export const notificationTypes = {
+  CAUTION: "caution",
+  INFORMATION: "information",
+  NEGATIVE: "negative",
+  POSITIVE: "positive",
+};
+
+/**
+ * The props for the Notification component.
+ */
+export type Props = {
+  /**
+   * The list of actions that the notification can perform.
+   */
+  actions?: {
+    appearance?: ButtonProps["appearance"];
```

Comment:
As discussed in stand-up, I guess we should allow for as little as possible at the beginning.

So let's limit that to `label` and `onClick` only for now. Ideally (if TS allows for that) we should only allow max 2 elements in actions array.

Reply:
Restricting size of array seems to be possible with TS tuples: https://stackoverflow.com/questions/65495285/typescript-restrict-maximum-array-length

So I guess something along those lines may work:


```ts

type NotificationAction = {
   label: string;
   onClick: () => void;
}

export type Props = {
   actions?: readonly [NotificationAction?, NotificationAction?];
   /* ... rest of the props */
}
```